tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -277,11 +285,16 @@ ROOT::Experimental::Internal::RPageSourceFile::CreateFromAnchor(const RNTuple &a
    if (!anchor.fFile)
       throw RException(R__FAIL("This RNTuple object was not streamed from a ROOT file (TFile or descendant)"));
 
-   // TODO(jblomer): Add RRawFile factory that create a raw file from a TFile. This may then duplicate the file
-   // descriptor (to avoid re-open).  There could also be a raw file that uses a TFile as a "backend" for TFile cases
-   // that are unsupported by raw file.
-   auto path = anchor.fFile->GetEndpointUrl()->GetFile();
-   auto pageSource = std::make_unique<RPageSourceFile>("", path, options);
+   std::unique_ptr<ROOT::Internal::RRawFile> rawFile;
+   // For local files, we want to open a new RRawFile to take advantage of the faster reading.
```

Comment:
I think we want to extend that to XRootD and HTTP, where RRawFile has its own plugins.